File: services/server/src/main/java/com/yaytsa/server/infrastructure/persistence/entity/PlayHistoryEntity.java
package com.yaytsa.server.infrastructure.persistence.entity;

import jakarta.persistence.*;
import java.time.OffsetDateTime;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "play_history")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class PlayHistoryEntity {

  @Id
  @GeneratedValue(strategy = GenerationType.UUID)
  private UUID id;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(
      name = "user_id",
      nullable = false,
      foreignKey = @ForeignKey(name = "fk_play_history_user"))
  private UserEntity user;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(
      name = "item_id",
      nullable = false,
      foreignKey = @ForeignKey(name = "fk_play_history_item"))
  private ItemEntity item;

  @Column(name = "started_at", nullable = false)
  private OffsetDateTime startedAt;

  @Column(name = "duration_ms", nullable = false)
  private Long durationMs;

  @Column(name = "played_ms", nullable = false)
  private Long playedMs = 0L;

  @Column(nullable = false)
  private Boolean completed = false;

  @Column(nullable = false)
  private Boolean scrobbled = false;

  @Column(nullable = false)
  private Boolean skipped = false;
}

File: yay-tsa-v2/infra-persistence/playback/src/main/kotlin/dev/yaytsa/persistence/playback/entity/PlayHistoryEntity.kt
package dev.yaytsa.persistence.playback.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.Instant
import java.util.UUID

@Entity
@Table(name = "play_history", schema = "core_v2_playback")
class PlayHistoryEntity(
    @Id
    val id: UUID = UUID.randomUUID(),
    @Column(name = "user_id", nullable = false)
    val userId: String = "",
    @Column(name = "item_id", nullable = false)
    val itemId: String = "",
    @Column(name = "started_at", nullable = false)
    val startedAt: Instant = Instant.now(),
    @Column(name = "duration_ms", nullable = false)
    val durationMs: Long = 0,
    @Column(name = "played_ms", nullable = false)
    val playedMs: Long = 0,
    @Column(nullable = false)
    val completed: Boolean = false,
    @Column(nullable = false)
    val scrobbled: Boolean = false,
    @Column(nullable = false)
    val skipped: Boolean = false,
)
